### PR TITLE
Proofread Stochastic Depth, SWA, and Augmentation Docstrings

### PR DESCRIPTION
**Pull request by mcneela**
Proofread Stochastic Depth, SWA, and Augmentation Docstrings

### CODE REVIEW

File: composer/algorithms/utils/augmentation_common.py [hunk 1]
```diff
@@ -11,28 +11,29 @@
...
             operate on batch tensors can safely call
             ``image_as_type(image, torch.Tensor)`` without additional error
             and type checking.
-        typ: type of the copied image. Must be :class:`PIL.Image.Image`
-            or :class:`torch.Tensor`
+        typ (:class:`torch.Tensor` | :class:`PIL.Image.Image`): type of the 
+            copied image. Must be :class:`PIL.Image.Image` or :class:`torch.Tensor`.
 
     Returns:
         A copy of ``image`` with type ``typ``
```
Comment (Landanjs) [inline composer/algorithms/utils/augmentation_common.py:30]: Do returns need types?

File: composer/algorithms/utils/augmentation_primitives.py [hunk 2]
```diff
@@ -129,8 +137,9 @@ def posterize(pil_img: Image.Image, level: float):
     .. seealso:: :func:`PIL.ImageOps.posterize`.
 
     Args:
-        pil_img (Image.Image): The image
-        level (float): The intensity, which should be on ``[0, 10]``
+        pil_img (:class:`Image.Image`): The image.
+        level (float): The intensity, which should
+            be in ``[0, 10]``
```
Comment (Landanjs) [inline composer/algorithms/utils/augmentation_primitives.py:142]: Is a period needed here?

File: composer/algorithms/swa/swa.py [hunk 3]
```diff
@@ -70,29 +71,29 @@ class SWA(Algorithm):
...
             anneal SWA learning rate. Note that updates are determined by the
             ``update_interval`` argument. For example, if ``anneal_steps = 10`` and
             ``update_interval = '1ep'``, then the SWA LR will be annealed once per epoch
             for 10 epochs; if ``anneal_steps = 20`` and ``update_interval = '8ba'``, then
             the SWA LR will be annealed once every 8 batches over the course of 160
-            batches (20 steps * 8 batches/step). Default = ``10``.
+            batches (20 steps * 8 batches/step). Default: ``10``.
         swa_lr (float, optional): The final learning rate to anneal towards with the SWA
-            LR scheduler. Set to ``None`` for no annealing. Default = ``None``.
+            LR scheduler. Set to ``None`` for no annealing. Default :``None``.
```
Comment (Landanjs) [inline composer/algorithms/swa/swa.py:96]:
```suggestion
            LR scheduler. Set to ``None`` for no annealing. Default: ``None``.
```

**Review by Landanjs — APPROVED** [review]
A few minor questions and suggestions, but LGTM! 

My main concern is `Image.Image` -> `PIL.Image.Image`, so the links in docs can render properly

File: composer/algorithms/utils/augmentation_primitives.py [hunk 4]
```diff
@@ -105,7 +113,7 @@ def autocontrast(pil_img: Image.Image, level: float = 0.0):
     .. seealso:: :func:`PIL.ImageOps.autocontrast`.
 
     Args:
-        pil_img (Image.Image): The image
+        pil_img (:class:`Image.Image`): The image.
```
Comment (Landanjs) [inline composer/algorithms/utils/augmentation_primitives.py:116]:
```suggestion
        pil_img (:class:`PIL.Image.Image`): The image.
```

I think `PIL.` is needed here (and all subsequent references) for the type link to be rendered properly ([see docs here](https://mosaicml-composer--999.com.readthedocs.build/en/999/api_reference/composer.algorithms.utils.augmentation_primitives.html?highlight=augmentation_primitives))

File: composer/algorithms/stochastic_depth/sample_stochastic_layers.py [hunk 5]
```diff
@@ -8,6 +8,12 @@ def _sample_drop(x: torch.Tensor, sample_drop_rate: float, is_training: bool):
     """Randomly drops samples from the input batch according to the `sample_drop_rate`.
```
Comment (growlix) [inline composer/algorithms/stochastic_depth/sample_stochastic_layers.py:8]: This file needs a module-level docstring ([example](https://github.com/mosaicml/composer/blob/01f671ed3c5cab2e26ec3e8bd201694331413027/composer/algorithms/colout/colout.py#L3)) and an [`__all__`](https://github.com/mosaicml/composer/blob/01f671ed3c5cab2e26ec3e8bd201694331413027/composer/algorithms/colout/colout.py#L28).

File: composer/algorithms/stochastic_depth/stochastic_depth.py [hunk 6]
```diff
@@ -38,7 +38,8 @@ def apply_stochastic_depth(model: torch.nn.Module,
                            drop_distribution: str = 'linear',
                            use_same_gpu_seed: bool = True,
                            optimizers: Optional[Union[Optimizer, Sequence[Optimizer]]] = None) -> torch.nn.Module:
-    """Applies Stochastic Depth (`Huang et al, 2016 <https://arxiv.org/abs/1603.09382>`_) to the specified model.
+    """Applies Stochastic Depth (`Huang et al, 2016 <https://arxiv.org/abs/1603.09382>`_)
```
Comment (growlix) [inline composer/algorithms/stochastic_depth/stochastic_depth.py:41]: File needs a module-level docstring and `__all__`.

File: composer/algorithms/stochastic_depth/stochastic_depth.py [hunk 7]
```diff
@@ -189,17 +199,19 @@ def __init__(self,
...
 
     def match(self, event: Event, state: State) -> bool:
         """Run on :attr:`~composer.core.event.Event.INIT`, as well as
-    :attr:`~composer.core.event.Event.BATCH_START` if ``drop_warmup > 0``.
+        :attr:`~composer.core.event.Event.BATCH_START` if ``drop_warmup > 0``.
 
         Args:
-            event (Event): The current event.
-            state (State): The current state.
+            event (:class:`Event`): The current event.
```
Comment (growlix) [inline composer/algorithms/stochastic_depth/stochastic_depth.py:213]: Type links not rendering. Also other algorithms don't seem to have docstrings for `match`, `apply`, and `find_unused_parameters`, so please confirm the convention with @ravi-mosaicml or @hanlint

File: composer/algorithms/stochastic_depth/stochastic_layers.py [hunk 8]
```diff
@@ -13,7 +13,9 @@ def _sample_bernoulli(probability: torch.Tensor,
                       use_same_gpu_seed: bool = False):
     """Gets a sample from a Bernoulli distribution.
 
-    Provides functionality to have different seeds across GPUs and to have the same set of seeds across GPUs.
+    Provides functionality to have different seeds
```
Comment (growlix) [inline composer/algorithms/stochastic_depth/stochastic_layers.py:16]: Module-level docstring and `__all__`.

**Review by growlix — COMMENTED** [review]
Some of these files need module-level docstrings and `__all__`, otherwise mostly just a few links not rendering and confirming conventions.

File: composer/algorithms/stochastic_depth/stochastic_depth.py [hunk 9]
```diff
@@ -131,26 +139,28 @@ class StochasticDepth(Algorithm):
...
+            ``"linear"`` linearly increases the drop rate across layer depth,
             starting with 0 drop rate and ending with ``drop_rate``. Default: ``"linear"``.
-        drop_warmup (str | Time | float, optional): A :class:`Time` object, time-string, or float
-            on [0.0; 1.0] representing the fraction of the training duration to linearly
-            increase the drop probability to `linear_drop_rate`. Default: ``0.0``.
+        drop_warmup (str | :class:`Time` | float, optional): A :class:`Time` object, 
+            time-string, or float on ``[0.0, 1.0]`` representing the fraction of the
+            training duration to linearly increase the drop probability to 
+            `linear_drop_rate`. Default: ``0.0``.
         use_same_gpu_seed (bool, optional): Set to ``True`` to have the same layers dropped
```
[the remark this answers is not in the document]
Comment (mcneela) [reply composer/algorithms/stochastic_depth/stochastic_depth.py:164]: Just trying to keep the line lengths uniform across each class/function docstring.

File: composer/algorithms/stochastic_depth/stochastic_depth.py [hunk 10]
```diff
@@ -189,17 +199,19 @@ def __init__(self,
 
     @property
     def find_unused_parameters(self) -> bool:
-        """DDP parameter to notify that parameters may not have gradients if it is dropped during the forward pass."""
+        """DDP parameter to notify that parameters may not
```
[the remark this answers is not in the document]
Comment (mcneela) [reply composer/algorithms/stochastic_depth/stochastic_depth.py:202]: Just trying to have the lines not jut out too far beyond the function signature.

File: composer/algorithms/utils/augmentation_common.py (hunk 1 above)
Comment (mcneela) [reply composer/algorithms/utils/augmentation_common.py:30]: No, I don't believe so? I didn't add return types here.

File: composer/algorithms/stochastic_depth/sample_stochastic_layers.py (hunk 5 above)
Comment (mcneela) [reply composer/algorithms/stochastic_depth/sample_stochastic_layers.py:13]: Ah gotcha, I just added one.

File: composer/algorithms/stochastic_depth/stochastic_depth.py (hunk 6 above)
Comment (mcneela) [reply composer/algorithms/stochastic_depth/stochastic_depth.py:46]: Added now.

File: composer/algorithms/stochastic_depth/stochastic_layers.py (hunk 8 above)
Comment (mcneela) [reply composer/algorithms/stochastic_depth/stochastic_layers.py:21]: Added now.

File: composer/algorithms/swa/swa.py (hunk 3 above)
Comment (mcneela) [reply composer/algorithms/swa/swa.py:96]: Good catch.

File: composer/algorithms/utils/augmentation_primitives.py (hunk 4 above)
Comment (mcneela) [reply composer/algorithms/utils/augmentation_primitives.py:116]: Got it, I didn't realize cross-linking was enabled for PIL. I'll add those from here on out.

File: composer/algorithms/utils/augmentation_primitives.py (hunk 2 above)
Comment (mcneela) [reply composer/algorithms/utils/augmentation_primitives.py]: Yes, good catch. I just added one.

**Comment by mcneela** [comment]
@hanlint Not sure why Jenkins is failing. The build works locally on my end.

File: composer/algorithms/stochastic_depth/stochastic_depth.py [hunk 11]
```diff
@@ -50,29 +56,32 @@ def apply_stochastic_depth(model: torch.nn.Module,
...
-            have each GPU drop a different set of layers. Only used
+            ``"linear"`` linearly increases the drop rate across layer depth,
+            starting with 0 drop rate and ending with ``drop_rate``. 
+            Default: ``"linear"``.
+        use_same_gpu_seed (bool, optional): Set to ``True`` to have the 
+            same layers dropped across GPUs when using multi-GPU training. 
+            Set to ``False`` to have each GPU drop a different set of layers. Only used
             with ``"block"`` stochastic method. Default: ``True``.
-        optimizers (torch.optim.Optimizer | Sequence[torch.optim.Optimizer], optional):
+        optimizers (:class:`torch.optim.Optimizer` | Sequence[:class:`torch.optim.Optimizer`], optional):
```
Comment (hanlint) [inline composer/algorithms/stochastic_depth/stochastic_depth.py:84]: only catching this now, but these are already resolving correctly without the need for a `:class:` marker, see: https://docs.mosaicml.com/en/latest/api_reference/composer.algorithms.stochastic_depth.stochastic_depth.html#composer.algorithms.stochastic_depth.stochastic_depth.apply_stochastic_depth

File: composer/algorithms/stochastic_depth/stochastic_depth.py (hunk 7 above)
Comment (hanlint) [reply composer/algorithms/stochastic_depth/stochastic_depth.py:219]: The original type links already render (https://docs.mosaicml.com/en/latest/api_reference/composer.algorithms.stochastic_depth.stochastic_depth.html#composer.algorithms.stochastic_depth.stochastic_depth.StochasticDepth.apply) so not changed are neccessary here.

File: composer/algorithms/utils/augmentation_common.py [hunk 12]
```diff
@@ -12,28 +12,29 @@
 
 def image_as_type(image: _InputImgT, typ: Type[_OutputImgT]) -> _OutputImgT:
     """Converts between :class:`torch.Tensor` and
-            :class:`PIL.Image.Image` image representations
+    :class:`PIL.Image.Image` image representations.
 
     Args:
-        image: a single image represented as a :class:`PIL.Image.Image` or
+        image (:class:`torch.Tensor` | :class:`PIL.Image.Image`): a single image
```
Comment (hanlint) [inline composer/algorithms/utils/augmentation_common.py:18]: `Tensor` should just work here?

File: composer/algorithms/utils/augmentation_primitives.py [hunk 13]
```diff
@@ -3,26 +3,27 @@
...
 
 Adapted from
-`AugMix: A Simple Data Processing Method to Improve Robustness and Uncertainty <https://github.com/google-research/augmix/blob/master/augmentations.py>`_.
+`AugMix: A Simple Data Processing Method to Improve Robustness and Uncertainty 
+<https://github.com/google-research/augmix/blob/master/augmentations.py>`_.
 
 Attributes:
-    AugmentationFn ((PIL.Image.Image, float) -> PIL.Image.Image): The type annotation for describing an
-        augmentation function.
+    AugmentationFn ((:class:`~PIL.Image.Image`, float) -> :class:`~PIL.Image.Image`):
```
Comment (hanlint) [inline composer/algorithms/utils/augmentation_primitives.py:14]: these already link, no `:class:` needed

File: composer/algorithms/utils/augmentation_primitives.py [hunk 14]
```diff
@@ -193,8 +206,9 @@ def translate_x(pil_img: Image.Image, level: float):
     """Shear an image horizontally.
 
     Args:
-        pil_img (Image.Image): The image.
-        level (float): The intensity, which should be on ``[0, 10]``.
+        pil_img (:class:`Image.Image`): The image.
```
Comment (hanlint) [inline composer/algorithms/utils/augmentation_primitives.py:209]: should be using `PIL.Image.Image` without the `:class:` marker.

**Review by hanlint — CHANGES_REQUESTED** [review]
Same feedback as on the `core` docstrings PR -- most of the `Args:` parameters don't need a `:class:` prefix to the type. Please check against the built docs to verify.

File: composer/algorithms/stochastic_depth/stochastic_depth.py (hunk 7 above)
Comment (ravi-mosaicml) [reply composer/algorithms/stochastic_depth/stochastic_depth.py:215]: It should not require the `:class:` notation here. I wonder what sphinx version are you on locally? Could be helpful to start with a clean virtual environment and re-install all dependencies; would that fix it?

Comment (mcneela) [reply composer/algorithms/stochastic_depth/stochastic_depth.py:215]: @ravi-mosaicml Yes, this is what the later PRs (e.g. #1073 ) were created to address. In that PR, I reverted all of these unneeded `:class:` annotations.

Comment (ravi-mosaicml) [reply composer/algorithms/stochastic_depth/stochastic_depth.py]: Sounds good. Going to merge this one in first, then do #1073.

**Pull request merged**